src/audio/SDL_audiocvt.c

```diff
@@ -157,15 +157,79 @@ static SDL_bool SDL_IsSupportedChannelCount(const int channels)
     return ((channels >= 1) && (channels <= 8));
 }
 
+SDL_bool SDL_ChannelMapIsBogus(const Uint8 *map, int channels)
+{
+    if (map) {
+        for (int i = 0; i < channels; i++) {
+            if (map[i] >= ((Uint8) channels)) {
+                return SDL_TRUE;
+            }
+        }
+    }
+    return SDL_FALSE;
+}
+
+SDL_bool SDL_ChannelMapIsDefault(const Uint8 *map, int channels)
+{
+    if (map) {
+        for (int i = 0; i < channels; i++) {
+            if (map[i] != i) {
+                return SDL_FALSE;
+            }
+        }
+    }
+    return SDL_TRUE;
+}
+
+// Swizzle audio channels. src and dst can be the same pointer. It does not change the buffer size.
+static void SwizzleAudio(const int num_frames, void *dst, const void *src, int channels, const Uint8 *map, int bitsize)
+{
+    #define CHANNEL_SWIZZLE(bits) { \
+        Uint##bits *tdst = (Uint##bits *) dst; /* treat as UintX; we only care about moving bits and not the type here. */ \
+        const Uint##bits *tsrc = (const Uint##bits *) src; \
+        if (src != dst) {  /* don't need to copy to a temporary frame first. */ \
+            for (int i = 0; i < num_frames; i++, tsrc += channels, tdst += channels) { \
+                for (int ch = 0; ch < channels; ch++) { \
+                    tdst[ch] = tsrc[map[ch]]; \
+                } \
+            } \
+        } else { \
+            Uint##bits tmp[32]; \
```

Comment:
You only need 16 here, right? Maybe this should be a constant?

Reply:
Yeah, good call.

Reply:
Fixed in c6465390f8213d87a7d1c59130b58c42bab95dcd.